src/main/java/com/minecolonies/coremod/colony/ColonyManager.java

```diff
@@ -536,6 +571,37 @@ private static void saveColonies()
         saveNeeded = false;
     }
 
+    public static boolean backupColonyData()
+    {
+        if (numWorldsLoaded > 0 && saveNeeded)
+        {
+            saveColonies();
+        }
+
+        @NotNull final File file = getSaveLocation();
+        @NotNull final File targetFile = getBackupSaveLocation(new Date());
+        if (!file.exists())
+        {
+            return true;
+        }
+        else if (targetFile.exists())
+        {
+            return false;
+        }
+
+        try
+        {
+            Files.copy(file.toPath(), targetFile.toPath());
+        }
+        catch (IOException e)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1166)